File: c/LAB/stack.c
#include<stdio.h>
#include<stdlib.h>

int *stack,top=-1,max;
void push(int ele)
{
     if(top==max-1)
    {
        printf("\nSTACK OVERFLOW");
        return;
    }

    //int n;
    //scanf("%d",&n);
    //stack[++top]=n;

    stack[++top]=ele;
}
int pop()
{
    if(top==-1)
    {
        printf("\nSTACK UNDERFLOW");
        return -1;
    }
    return(stack[top--]);
}
void display()
{   
    if(top==-1)
    {
        printf("\nSTACK UNDERFLOW");
        return ;
    }
    printf("\nThe elements are\n");
    for(int i=top;i>=0;i--)
    {
       printf("%d\t",stack[i]);
    }
    return;
}
void palindrome()
{   top=-1;
    int n,num,copy;
    printf("\nENter the number:");
    scanf("%d",&num);
    copy=num;
    while(copy!=0)
    {
        push(copy%10);
        copy=copy/10;
    }
    while(num!=0)
    {
        if(num%10!=pop())
        {
            printf("\nNOT PALINDROME\n");
            top=-1;
            return;
        }
        num=num/10;
    }
    printf("\nPALINDROME");

}
void main()
{   printf("\nEnter the size");
    scanf("%d",&max);
    stack=(int*)malloc(max*sizeof(int));
    int n,ele;
    while(1)
    {   
        printf("\n1-push 2-pop 3-display 4-palindrome 5-exit");
        scanf("%d",&n);
        switch(n){
            case 1:printf("\nEnter the element:");
                   scanf("%d",&ele);
                   push(ele);
                    break;
            case 2:ele=pop();
                    if(ele!=-1)
                    {                    
                    printf("\npoped element is %d",ele);
                    }   
                   break;
            case 3:display();break;
            case 4:palindrome();break;
            case 5:exit(0);
            default:printf("\nINVALID");

        }
    }
}

File: java/LAB/stack.java
import java.util.Scanner;
class stack1{
    Scanner s=new Scanner(System.in);
    
        int max=10,top=-1;
        int stack[]=new int[max];
        void push()
        {
            if(top==max-1) System.out.println("STACK OVERFLOW");
            else {System.out.println("Enter the element");stack[++top]=s.nextInt();}
        }
        void pop()
        {
            if(top==-1){System.out.println("STACK UNDERFLOW");}
            else{System.out.println("The deleted element is "+stack[top--]);}
        }
        void display()
        {
            if(top==-1){System.out.println("STACK IS EMPTY");}
            else{for(int i=top;i>=0;i--)System.out.print(stack[i] +"\t");System.out.println();}
        }
}
public class stack{
        public static void main(String srgs[])
        { //  Scanner s=new Scanner(System.in);
            stack1 obj=new stack1();

            while(true)
            {
            System.out.println("1-push\t2-pop\t3-display\t4-exit");
            int n=obj.s.nextInt();
            switch (n) {
                case 1:obj.push();
                    
                    break;
                case 2:obj.pop();
                    
                    break;
                case 3:obj.display();
                    
                    break;
                case 4:System.exit(0);
                default:System.out.println("Invalid choice");
                    break;
            }
        }
    }
}
